backend/src/test/java/com/example/subjecthub/SubjectServiceControllerTests.java

```diff
@@ -138,6 +146,134 @@ public void testGetSubjectsFuzzySearch() throws Exception {
     }
 
     /**
+     * need 'create user' tests above comment tests to create user and get it's id for the addComment test
+     */
+
+    @Test
+    public void testAddComment() throws Exception {
+        // creates dud subject then adds a comment
+        Subject s = createSubject("test", "ABCDEF");
+        Long u_id = Long.parseLong("1"); //alternatively (reccomended), create user and get it's id
+        Long s_id = s.getId();
+        String m = "tasty";
+
+        AddCommentRequest addCommentRequest = new AddCommentRequest(u_id, m);
+
+        // Checks that new subject appears in list
+        mockMvc.perform(post("/api/universities/university/1/subjects/subject/"+s_id+"/comments/comment/add")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content("{\"userId\":\""+addCommentRequest.getUserId()+
```

Comment:
No need to change this, but for future reference I've created a TestUtils class which has a toJson method to convert our objects to json strings for the purposes of testing.

Reply:
Thanks :D